src/postprocessors/NekPostprocessor.C

```diff
@@ -24,11 +24,15 @@ InputParameters
 NekPostprocessor::validParams()
 {
   InputParameters params = GeneralPostprocessor::validParams();
+
+  params.addParam<bool>("flowMesh", false, "Only compute postprocessor for flow mesh, not solid mesh");
```

Comment:
What do you think about an enumeration instead? We might want to integrate over both the fluid & solid mesh for porous applications. Something like this:

```
domain = 'fluid solid '
domain = 'fluid'
domain = 'solid'
```

Reply:
Yes that can work and add more capability. I know we can do `domain = 'fluid solid '` and `domain = 'fluid'`, but I'm not confident on how to do `domain = 'solid'`. 

For  `domain = 'fluid'` we can just pull the flowMesh().
- This is same as having the parameter `flowMesh=True` right now in what I implemented. No problem here.

For  `domain = 'fluid solid'` we can just pull the temperatureMesh() (entireMesh()=temperatureMesh() for this case)
- This is same as having the parameter `flowMesh=False` right now in what I implemented. No problem here.

But for `domain = 'solid'` we currently don't have a mesh in Cardinal that points to the solid mesh only, and I'm not sure how to do this. Could you help me with this? Unless we can just do `solidMesh() = temperatureMesh() - fluidMesh()`, that seems too easy.

Reply:
You're right about the element ordering, and yes I will follow your suggestions.

For `domain = 'fluid solid'` with `field=velocity_x` , it doesn't seg fault because the solid mesh still has velocity(x,y,z) variables they are all just set to zero. So there's no seg faulting.

Reply:
Good to know, thanks! 

I think we will also want to avoid having 2 versions of every function - it can become more effort to maintain if we ever need to change the behavior of those functions. I'd suggest just deleting the previous versions of the functions and keeping the new versions with the extra parameter. Perhaps having this extra parameter be a [default argument](https://en.cppreference.com/w/cpp/language/default_arguments) can help with not needing to touch other parts of the code base.

FYI you will need to rebase this branch off the latest devel, I merged some other things earlier this morning.

Reply:
Okay thanks!